plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java

```diff
@@ -360,16 +361,17 @@ private static Optional<PartitionData> getPartitionData(List<PartitionColumn> co
             PartitionColumn column = columns.get(i);
             Block block = page.getBlock(column.getSourceChannel());
             Type type = column.getSourceType();
+            org.apache.iceberg.types.Type icebergType = toIcebergType(type);
```

Comment:
Can we get the Iceberg type from existing fields? e.g. `Schema outputSchema`

Reply:
Yes, After making `getPartitionData` a non-static method, we can use `outputSchema` to get the iceberg type.
``` java
 org.apache.iceberg.types.Type icebergType = outputSchema.findType(column.getField().sourceId());
````